src/protocols/TRXLooper.cpp

```diff
@@ -713,215 +301,128 @@ int TRXLooper::StreamRx(lime::complex16_t* const* dest, uint32_t count, SDRDevic
-    {
-        if (!mTx.stagingPacket)
         {
-            mTx.stagingPacket = SamplesPacketType::ConstructSamplesPacket(
-                mTx.memPool->Allocate(outputPktSize), samplesInPkt * packetsToBatch, sizeof(complex32f_t));
-            if (!mTx.stagingPacket)
-                break;
-            mTx.stagingPacket->Reset();
-            mTx.stagingPacket->timestamp = ts;
-            mTx.stagingPacket->useTimestamp = useTimestamp;
+            return 0;
         }
 
-        int consumed = mTx.stagingPacket->push(src, samplesRemaining);
-        src[0] += consumed;
-        if (useChannelB)
-            src[1] += consumed;
-
-        samplesRemaining -= consumed;
-        ts += consumed;
-
-        if (mTx.stagingPacket->isFull() || flush)
-        {
-            if (samplesRemaining == 0)
-                mTx.stagingPacket->flush = flush;
-
-            if (!mTx.fifo->push(mTx.stagingPacket))
-                break;
-            mTx.stagingPacket = nullptr;
-        }
-    }
-    return count - samplesRemaining;
-}
-
-int TRXLooper::StreamTx(const lime::complex16_t* const* samples, uint32_t count, const SDRDevice::StreamMeta* meta)
-{
-    const bool useChannelB = mConfig.channels.at(lime::TRXDir::Tx).size() > 1;
-    const bool useTimestamp = meta ? meta->useTimestamp : false;
-    const bool flush = meta && meta->flush;
-    int64_t ts = meta ? meta->timestamp : 0;
-
-    int samplesRemaining = count;
-    const int samplesInPkt = mTx.samplesInPkt;
-    const int packetsToBatch = mTx.packetsToBatch;
-    const int32_t outputPktSize = SamplesPacketType::headerSize + packetsToBatch * samplesInPkt * sizeof(complex16_t);
-
-    if (mTx.stagingPacket && mTx.stagingPacket->timestamp + mTx.stagingPacket->size() != meta->timestamp)
-    {
-        if (!mTx.fifo->push(mTx.stagingPacket))
-            return 0;
         mTx.stagingPacket = nullptr;
     }
 
-    const lime::complex16_t* src[2] = { samples[0], useChannelB ? samples[1] : nullptr };
-    while (samplesRemaining)
+    const T* src[2] = { samples[0], useChannelB ? samples[1] : nullptr };
+    while (samplesRemaining > 0)
     {
         if (!mTx.stagingPacket)
         {
             mTx.stagingPacket = SamplesPacketType::ConstructSamplesPacket(
-                mTx.memPool->Allocate(outputPktSize), samplesInPkt * packetsToBatch, sizeof(complex16_t));
+                mTx.memPool->Allocate(outputPktSize), samplesInPkt * packetsToBatch, sizeof(T));
+
             if (!mTx.stagingPacket)
+            {
                 break;
+            }
+
             mTx.stagingPacket->Reset();
-            mTx.stagingPacket->timestamp = ts;
-            mTx.stagingPacket->useTimestamp = useTimestamp;
+            mTx.stagingPacket->metadata.timestamp = ts;
+            mTx.stagingPacket->metadata.waitForTimestamp = useTimestamp;
         }
 
         int consumed = mTx.stagingPacket->push(src, samplesRemaining);
         src[0] += consumed;
         if (useChannelB)
+        {
```

Comment:
please stop adding these unnecessary braces for one liners. It does not help readablility (in fact worsens it, functions no longer fit into screen due to all the wasted space), and adds noise to the PRs.